command/agent/agent.go

```diff
@@ -1165,6 +1165,9 @@ func (a *Agent) Shutdown() error {
 		if err := a.client.Shutdown(); err != nil {
 			a.logger.Error("client shutdown failed", "error", err)
 		}
+
+		// Task API must be closed separately
+		a.builtinServer.Shutdown()
```

Comment:
It probably doesn't matter unless we block on graceful shutdown (maybe a future PR), but given the client shutdown can take a little bit, should the Task API shut down before the rest of the client so that requests aren't being served?